daml-lf/kv-support/src/main/scala/com/daml/lf/kv/transactions/TransactionConversions.scala

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.kv.transactions
+
+import com.daml.lf.transaction.{
+  NodeId,
+  TransactionCoder,
+  TransactionOuterClass,
+  VersionedTransaction,
+}
+import com.daml.lf.value.ValueCoder
+
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+
+object TransactionConversions {
+
+  def encodeTransaction(
+      tx: VersionedTransaction
+  ): Either[ValueCoder.EncodeError, RawTransaction] =
+    TransactionCoder
+      .encodeTransaction(TransactionCoder.NidEncoder, ValueCoder.CidEncoder, tx)
+      .map(transaction => RawTransaction(transaction.toByteString))
+
+  def decodeTransaction(
+      rawTx: RawTransaction
+  ): Either[ValueCoder.DecodeError, VersionedTransaction] = {
+    val transaction = TransactionOuterClass.Transaction.parseFrom(rawTx.byteString)
+    TransactionCoder
+      .decodeTransaction(
+        TransactionCoder.NidDecoder,
+        ValueCoder.CidDecoder,
+        transaction,
+      )
+  }
+
+  def encodeTransactionNodeId(nodeId: NodeId): RawTransaction.NodeId =
+    RawTransaction.NodeId(nodeId.index.toString)
+
+  def decodeTransactionNodeId(transactionNodeId: RawTransaction.NodeId): NodeId =
+    NodeId(transactionNodeId.value.toInt)
+
+  def extractTransactionVersion(rawTransaction: RawTransaction): String =
+    TransactionOuterClass.Transaction.parseFrom(rawTransaction.byteString).getVersion
+
+  def extractNodeId(rawTransactionNode: RawTransaction.Node): RawTransaction.NodeId =
+    RawTransaction.NodeId(
+      TransactionOuterClass.Node.parseFrom(rawTransactionNode.byteString).getNodeId
+    )
+
+  def reconstructTransaction(
```

Comment:
for the rest, I can accept no tests at least for this PR but this definitely should have some.

Reply:
This one has a test in `TransactionConversionsSpec`. However, this is probably the right moment to add more but I'm moving things one-to-one not to make the PR bigger. So, I can add tests here. Add tests in a PR that is gonna be merged before this one, or after merging this one. WDYT, @cocreature?

Reply:
Ah missed that, I’d add a TODO in this PR and address it in a later PR.

Reply:
can we add the todo before merging please

Reply:
I will add more tests before merging and ask Samir to review.